### PR TITLE
Support custom classes for Message Feed

### PR DESCRIPTION
**Pull request by swarna04**
## Description

Ref: [MOB-19181]

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPMessaging/Sources/PropositionItem.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    public let uniqueId: String
+
+    /// PropositionItem schema string
+    public let schema: String
+
+    /// PropositionItem data content e.g. html or plain-text string or string containing image URL, JSON string
+    public let content: String
+
+    /// Weak reference to Proposition instance
+    weak var proposition: Proposition?
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/PropositionItem.swift:29]: are we using this or is meant to be potentially used by the developer?

File: AEPMessaging/Sources/PropositionItem.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            let jsonData = codableContent.dictionaryValue,
+            let encodedData = try? JSONSerialization.data(withJSONObject: jsonData),
+            let contentString = String(data: encodedData, encoding: .utf8)
+        {
+            content = contentString
+            return
+        }
+        throw DecodingError.typeMismatch(PropositionItem.self,
+                                         DecodingError.Context(codingPath: decoder.codingPath,
+                                                               debugDescription: "PropositionItem content is not of an expected type."))
```
Comment (sbenedicadb) [inline AEPMessaging/Sources/PropositionItem.swift:63]: nit: i think it's cleaner to use else-ifs here rather than separate ifs with multiple returns.

**Review by sbenedicadb — APPROVED** [review]
looks great with a couple minor comments

File: AEPMessaging/Sources/PropositionItem.swift (hunk 1 above)
Comment (swarna04) [reply AEPMessaging/Sources/PropositionItem.swift:29]: Mostly by the developer.

**Pull request merged**